x-pack/plugins/security_solution/common/endpoint/types/actions.ts

```diff
@@ -224,3 +221,12 @@ export interface ActionDetails {
 export interface ActionDetailsApiResponse {
   data: ActionDetails;
 }
+export interface ActionListApiResponse {
+  page: number | undefined;
+  pageSize: number | undefined;
+  startDate: string | undefined;
+  endDate: string | undefined;
+  userIds: string[] | undefined; // users that requested the actions
+  commands: string[] | undefined; // type of actions, should have at least one always
```

Comment:
The reason I'd be fine with proactively returning the filtering properties is to facilitate a "forgetful" frontend UI. For instance, if the frontend state didn't store/remember what the filter inputs were on every send, and could rebuild from the response. Or if an action-search could be initiated from another page. For instance, say you're looking at an Endpoint details section, it has like a "last action" section where you can click parts of it. Like say you click on the command "isolate" and it sends you to the list of all recent "isolate" commands. This will probably send you to a list-results page. The list page can rebuild the query state by the response.

Or something to that effect. But essentially, I don't like intentionally hiding information that a backend has (as long as its not something internal we intend to change). The more we expose, the less the backend has to _ever_ change as the frontend design or flow changes. The fewer calls one has to make to get more info, etc etc.

Reply:
I'm ok if you all want to return the filtering criteria. Note that it could bloat the response message size unnecessarily as some filtering criteria could be long. Also - we (UI) already have the information we sent with the API available, so certainly I don't see a use case where we would need it on the response.

Ok - I'll defer to whatever you all think is best then. but: I still think `total` is missing

Reply:
I've added `total` to the response

Reply:
I'm also in favor of keeping the response in sync with the request query, that's the reason I'm returning the requested query keys.